staging/src/k8s.io/component-base/logs/registry.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logs
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/go-logr/logr"
+)
+
+var logRegistry *LogFormatRegistry = NewLogFormatRegistry()
+
+// LogFormatRegistry store klog format registry
+type LogFormatRegistry struct {
+	registry map[string]logr.Logger
+	mu       sync.Mutex
+}
+
+func NewLogFormatRegistry() *LogFormatRegistry {
+	return &LogFormatRegistry{
+		registry: make(map[string]logr.Logger),
+		mu:       sync.Mutex{},
```

Comment:
@mtaufen should register be thread safe?

Reply:
IMO this is something that should be set up very early in the lifecycle, and not from multiple threads, but it's also easy to add a mutex and I don't think there's a performance issue, so why take the chance?